bokehjs/src/coffee/core/has_props.coffee

```diff
@@ -266,40 +287,30 @@ class HasProps extends Backbone.Model
     # make this a no-op, we sync the whole document never individual models
     return options.success(model.attributes, null, {})
 
-  defaults: -> { }
+  defaults: -> throw new Error("don't use HasProps.defaults anymore")
 
   # TODO remove this, for now it's just to help find nonserializable_attribute_names we
   # need to add.
   serializable_in_document: () -> true
 
-  # returns a list of those names which should not be included
-  # in the Document and should not go to the server. Subtypes
-  # should override this. The result will be cached on the class,
-  # so this only gets called one time on one instance.
-  nonserializable_attribute_names: () -> []
-
-  _get_nonserializable_dict: () ->
-    if not @constructor._nonserializable_names_cache?
-      names = {}
-      for n in @nonserializable_attribute_names()
-        names[n] = true
-      @constructor._nonserializable_names_cache = names
-    @constructor._nonserializable_names_cache
-
   attribute_is_serializable: (attr) ->
-    (attr not of @_get_nonserializable_dict()) and (attr of @attributes)
+    if attr == "id"
```

Comment:
This is (and other similar instances) is ugly. I may need to to  add `id` to `@props`.